File: to50/AGC23/mainA.rb
n = gets.chomp.to_i
a = gets.chomp.split(" ").map(&:to_i)
s = [0]
n.times do |i|
    s << s[i] + a[i]
end

h = Hash.new(0)
ans = 0

s.each do |i|
    h[i] += 1
end

h.each do |key, value|
    ans += value * (value-1) / 2 if 2 <= value
end
puts ans

File: from101to150/ARC108/mainA.kt
fun main() {
    val (s: Long, p: Long) = readLine()!!.split(" ").map(String::toLong)
    var n: Long = 1L
    var ans: Boolean = false
    while (n*n <= p) {
        var m: Long = s-n
        if (m <= 0) continue
        if (n*m == p) {
            ans = true
            break
        }
        n++
    }
    println(if(ans) "Yes" else "No")
}